### PR TITLE
Refactor localfile implementation to use a local RS endpoint client

### PR DESCRIPTION
**Pull request by basiliskus**
# Refactor localfile implementation to use a local RS endpoint client

Removed `LocalFileOrderSender` and instead created a `RSEndpointClient` interface with two implementations: one that calls RS `ReportStreamEndpointClient` and a local one `LocalEndpointClient` that doesn't. This way we can abstract the dependency with RS in multiple places in our code, which is handy for PR 746 which requires calling the RS history API

## Issue

#672 
## Checklist

- [x] I have added tests to cover my changes

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import gov.hhs.cdc.trustedintermediary.etor.RSEndpointClient;
+import gov.hhs.cdc.trustedintermediary.external.reportstream.ReportStreamEndpointClientException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import javax.annotation.Nonnull;
+
+public class LocalEndpointClient implements RSEndpointClient {
```
Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java:11]: Hmmm, I'm thinking of renaming this to `MockRsEndpointClient`.  I don't want this class to be confused for a locally running (in Docker or whatever) version of ReportStream.

Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java:11]: Add JavaDocs.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package gov.hhs.cdc.trustedintermediary.etor;
+
+import gov.hhs.cdc.trustedintermediary.external.reportstream.ReportStreamEndpointClientException;
+import javax.annotation.Nonnull;
+
+public interface RSEndpointClient {
```
Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java:6]: Add JavaDocs.

**Review by halprin — COMMENTED** [review]
Looks like some fun changes have gone in while I was gone.  Good work.  The JavaDocs comments are required, please.  You can take or leave the rename comment.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+    String getRsToken() throws ReportStreamEndpointClientException;
+
+    String requestWatersEndpoint(@Nonnull String body, @Nonnull String bearerToken)
+            throws ReportStreamEndpointClientException;
+
+    String requestHistoryEndpoint(@Nonnull String submissionId, @Nonnull String bearerToken)
```
Comment (jcrichlake) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java:13]: Can we take out these @Nonnull annotations? They apparently only work 1 call down and only within one's IDE.  Feels a bit like giving folks a false sense of security which could lead to issues.

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java:13]: That sounds good to me

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java (hunk 1 above)
Comment (tjohnson7021) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java:11]: I second that idea.

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java:11]: I'm ok with that. I was trying to follow the naming convention used for `LocalFileOrderSender`, but I agree `MockRsEndpointClient` it's clearer. I'll change it

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java:11]: Good catch. Will do

**Review by tjohnson7021 — APPROVED** [review]

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java (hunk 3 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java]: Done

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java]: Renamed

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/localfile/LocalEndpointClient.java]: Done

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java (hunk 2 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/RSEndpointClient.java:9]: Done

**Review by halprin — APPROVED** [review]
Ship it!

**Pull request merged**